File: apps/admin-dashboard/src/features/auth/api/authApi.ts
import { httpClient } from '../../../shared/lib/httpClient';
import type { LoginResponse, AuthenticatedUser } from '../../../shared/types/api';

export interface RegisterInput {
  email: string;
  password: string;
  fullName: string;
  familyName?: string;
}

export interface LoginInput {
  email: string;
  password: string;
}

export const authApi = {
  register(input: RegisterInput): Promise<AuthenticatedUser> {
    return httpClient<AuthenticatedUser>('/auth/register', {
      method: 'POST',
      body: input,
      skipAuth: true,
    });
  },

  login(input: LoginInput): Promise<LoginResponse> {
    return httpClient<LoginResponse>('/auth/login', {
      method: 'POST',
      body: input,
      skipAuth: true,
    });
  },

  logout(refreshToken: string): Promise<void> {
    return httpClient<void>('/auth/logout', {
      method: 'POST',
      body: { refreshToken },
    });
  },
};

File: apps/parent-app/lib/features/authentication/api/auth_api.dart
import '../../../core/network/api_client.dart';

class AuthApi {
  AuthApi(this._client);

  final ApiClient _client;

  Future<Map<String, dynamic>> login({required String email, required String password}) {
    return _client.post('/auth/login', data: {'email': email, 'password': password}, skipAuth: true);
  }

  Future<Map<String, dynamic>> register({
    required String fullName,
    required String email,
    required String password,
  }) {
    return _client.post(
      '/auth/register',
      data: {'fullName': fullName, 'email': email, 'password': password},
      skipAuth: true,
    );
  }

  Future<void> logout(String refreshToken) {
    return _client.post('/auth/logout', data: {'refreshToken': refreshToken});
  }
}
